pkg/controllers/nodetemplate/controller.go

```
@@ -96,17 +117,42 @@ func (c *Controller) resolveSubnets(ctx context.Context, nodeTemplate *v1alpha1.
 	return nil
 }
 
-func (c *Controller) resolveSecurityGroup(ctx context.Context, nodeTemplate *v1alpha1.AWSNodeTemplate) error {
+func (c *Controller) resolveSecurityGroups(ctx context.Context, nodeTemplate *v1alpha1.AWSNodeTemplate) error {
 	securityGroupIds, err := c.securityGroupProvider.List(ctx, nodeTemplate)
 	if err != nil {
 		return err
 	}
+	if len(securityGroupIds) == 0 {
+		nodeTemplate.Status.SecurityGroups = nil
+		logging.FromContext(ctx).Errorf("no securityGroup exist given constraints")
```

We should not just log an error here, I think we want to return an error